src/freertos_drivers/esp32/Esp32WiFiManager.cxx

```diff
@@ -0,0 +1,614 @@
+        VERIFY_CACHED_VALUE(hubPort_, hubPort)
+        VERIFY_CACHED_VALUE(hubServiceName_, hubServiceName)
+        VERIFY_CACHED_VALUE(tcpClientMode_, tcpClientMode)
+        VERIFY_CACHED_VALUE(tcpmDNSServiceName_, tcpmDNSServiceName)
+        VERIFY_CACHED_VALUE(tcpAutoHostName_, tcpAutoHostName)
+        VERIFY_CACHED_VALUE(tcpManualHostName_, tcpManualHostName)
+        VERIFY_CACHED_VALUE(tcpManualPort_, tcpManualPort)
+    }
+
+    // verify settings provided to make sure they are valid
+    if (tcpClientMode_ != SocketClientParams::SearchMode::AUTO_ONLY)
+    {
+        // if the manual hostname is not provided (or is blank) or we do not
+        // have a valid manual port force settings to AUTO_ONLY.
+        if (!tcpManualHostName_.length() || tcpManualPort_ == 0)
+        {
+            LOG(WARNING, "Missing manual hub hostname or port, "
+                "switching to AUTO_ONLY mode");
+            tcpClientMode_ = SocketClientParams::SearchMode::AUTO_ONLY;
+            // record the updated settings.
+            link_config.search_mode().write(fd, tcpClientMode_);
+        }
+    }
+
+    // if this is the first time we are loaded start the WiFi system.
+    if (initial_load)
+    {
+        startWiFiSystem();
+    }
+
+    // if the configuration settings have changed, restart the WiFi manager
+    // task to pick up the updated settings.
+    if (configUpdated)
+    {
+        stopWiFiTask();
+        startWiFiTask();
+    }
+
+    return ConfigUpdateListener::UpdateAction::UPDATED;
+}
+
+void Esp32WiFiManager::factory_reset(int fd)
+{
+    LOG(VERBOSE, "Esp32WiFiManager::factory_reset(%d)", fd);
+    // General WiFi configuration settings
+    CDI_FACTORY_RESET(cfg_.wifi_sleep);
+
+    // Hub specific configuration settings
+    CDI_FACTORY_RESET(cfg_.hub_config().enabled);
+    CDI_FACTORY_RESET(cfg_.hub_config().port);
+    cfg_.hub_config().service_name().write(fd,
+        TcpDefs::MDNS_SERVICE_NAME_GRIDCONNECT_CAN_TCP);
+
+    // node link configuration settings
+    CDI_FACTORY_RESET(cfg_.link_config().search_mode);
+    CDI_FACTORY_RESET(cfg_.link_config().reconnect);
+
+    // node link manual configuration settings
+    cfg_.link_config().manual_address().ip_address().write(fd, "");
+    CDI_FACTORY_RESET(cfg_.link_config().manual_address().port);
+
+    // node link automatic configuration settings
+    cfg_.link_config().auto_address().service_name().write(fd,
+        TcpDefs::MDNS_SERVICE_NAME_GRIDCONNECT_CAN_TCP);
+    cfg_.link_config().auto_address().host_name().write(fd, "");
+
+    // node link automatic last connected node address
+    cfg_.link_config().last_address().ip_address().write(fd, "");
+    CDI_FACTORY_RESET(cfg_.link_config().last_address().port);
+
+    // reconnect to last connected node
+    CDI_FACTORY_RESET(cfg_.link_config().reconnect);
+}
+
+void Esp32WiFiManager::startWiFiSystem()
+{
+    // if we are to manage the entire WiFi stack start and configure it here
+    if (manageWiFi_)
```

Comment:
could you invert this condition to save on indent?

Reply:
dropping this and moving the call to be guarded by the value in apply_configuration instead with the sleep mode moved into the same (when config updated it will get set, after wifi start)